evennia/utils/utils.py

```diff
@@ -472,6 +472,31 @@ def iter_to_str(iterable, sep=",", endsep=", and", addquote=False):
 list_to_string = iter_to_str
 iter_to_string = iter_to_str
 
+re_empty = re.compile("\n\s*\n")
+
+def strip_extra_whitespace(text, max_linebreaks=1, max_spacing=2):
+    """
+    Removes extra sequential whitespace in a block of text. This will also remove any trailing
+    whitespace at the end.
+
+    Args:
+        text (str):   A string which may contain excess internal whitespace.
+        
+    Keyword args:
+        max_linebreaks (int):  How many linebreak characters are allowed to occur in a row.
+        max_spacing (int):     How many spaces are allowed to occur in a row.
+
```

Comment:
I think it would be useful with an `Example:` block here: 

```
Example:
::
    limit_whitespace_in_string("""
         A long    string     with whitespaces 


         A line separated      by   many line breaks 
         """, max_linebreaks=2, max_spaces=1)
becomes 
:: 
    """
    A long string with whitespaces
          
    A line separated by many line breaks"""
```

Reply:
I couldn't figure out how to write an effective and legible example that didn't break the docstring.

Reply:
Ok, I'll investigate.